flink-connectors/flink-connector-kafka-0.11/src/main/java/org/apache/flink/streaming/connectors/kafka/internal/TransactionalIdsGenerator.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.connectors.kafka.internal;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * Class responsible for generating transactional ids to use when communicating with Kafka.
+ *
+ * <p>It guarantees that:
+ * - generated ids to use will never clash with ids to use from different subtasks
+ * - generated ids to abort will never clash with ids to abort from different subtasks
+ * - generated ids to use will never clash with ids to abort from different subtasks
+ *
+ * <p>In other words, any particular generated id will always be assigned to one and only one subtask.
+ */
+public class TransactionalIdsGenerator {
```

Comment:
It is better to use the unordered list tag, e.g.,
```
  * <ul>
  * 	<li>generated ids to use will never clash with ids to use from different subtasks</li>
  * 	<li>generated ids to abort will never clash with ids to abort from different subtasks</li>
  * 	<li>generated ids to use will never clash with ids to abort from different subtasks</li>
  * </ul>
```
Otherwise it is rendered like this:
![image](https://user-images.githubusercontent.com/1681921/32487045-c97ffe8a-c3a8-11e7-95e8-f3ee127072b9.png)

Reply:
ops, forgot about it

Reply:
another nitpick 😉: `<li>` tags are usually not closed in Javadoc: http://www.oracle.com/technetwork/articles/java/index-137868.html